.github/workflows/ci-cd.yml

```
@@ -119,54 +142,79 @@
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.Semantic-Release.outputs.tag }}
+
+      - name: setup python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install using TestPyPI
+        run: |
+          pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ readii
```

yamllint

[error] no new line character at the end of file

(new-line-at-end-of-file)